src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -458,6 +461,17 @@ public static void ClearPool(SqlConnection connection)
             }
         }
 
+        private void CleanOutTransactionJunk()
+        {
+            if (InnerConnection.Pool != null)
+            {
+                if (this.CurrentTransaction != null)
+                {
+                    this.CurrentTransaction.Dispose();
+                }
+                this.BeginTransaction(IsolationLevel.ReadCommitted).Dispose();
```

Comment:
it will send 2 request to server: `TM_BEGIN_XACT` and `TM_ROLLBACK_XACT`, and it does not receive response from server for those requests. The byte spec is described on https://msdn.microsoft.com/en-us/library/dd339887.aspx